engine/server/session.go

```diff
@@ -1044,6 +1044,11 @@ func (srv *Server) serveSessionAttachables(w http.ResponseWriter, r *http.Reques
 	}
 
 	ctx = client.daggerSession.withShutdownCancel(ctx)
+
+	// Disable collecting otel metrics on these grpc connections for now. We don't use them and
+	// they add noticeable memory allocation overhead, especially for heavy filesync use cases.
+	ctx = trace.ContextWithSpan(ctx, trace.SpanFromContext(nil)) //nolint:staticcheck // we have to provide a nil context...
```

Comment:
cc @vito, just want to confirm we are for sure not using any of these spans from buildkit session stuff

Reply:
Oh yeah it's confusing, having a span in the context triggered this code in buildkit: https://github.com/moby/buildkit/blob/c234dd97ba77033c55b844e59b4ab1122e7977f1/session/grpc.go#L51-L57

But it sets up a grpc StatsHandler which ends up collecting otel metrics.

Not sure if there's a reason span+metrics are tied together this way, but we don't want any of it so 🤷‍♂️